### PR TITLE
comodoro: add module

### PR DESCRIPTION
**Pull request by soywod**
### Description

**Comodoro, a CLI to manage your time**. Key features:

- Centralized server timer controllable by multiple clients at the same time
- Multi protocols (only TCP for now, but you can build your own)
- Cycles customizable via config file (Pomodoro style, 52/17 style, custom)
- Server and timer hooks customizable via config file

Website & doc: https://pimalaya.org/comodoro/

Sources: https://github.com/soywod/comodoro

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

@toastal

### CODE REVIEW

File: modules/programs/comodoro.nix [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      protocols;
+  };
+
+in {
+  meta.maintainers = with lib.hm.maintainers; [ soywod ];
+
+  options = {
+    programs.comodoro = {
+      enable =
+        lib.mkEnableOption "Enable Comodoro, the CLI to manage your time.";
```
Comment (sumnerevans) [inline modules/programs/comodoro.nix:22]: `mkEnableOption` automatically adds "Enable " to the front and "." to the end.

```suggestion
        lib.mkEnableOption "Comodoro, the CLI to manage your time";
```

File: modules/programs/comodoro.nix [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        default = { };
+        description = ''
+          Comodoro configuration.
+          See <link xlink:href="https://pimalaya.org/comodoro/cli/configuration/"/> for supported values.
+        '';
+      };
+    };
+
+    services.comodoro = {
+      enable = lib.mkEnableOption "Enable the Comodoro server.";
```
Comment (sumnerevans) [inline modules/programs/comodoro.nix:35]:
```suggestion
      enable = lib.mkEnableOption "the Comodoro server";
```

File: modules/programs/comodoro.nix [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            Use configuration from the given preset as defined in the configuration file.
+          '';
+        };
+
+        protocols = lib.mkOption {
+          type = with lib.types; nonEmptyListOf nonEmptyStr;
+          description = ''
+            Define protocols the server should use to accept requests.
+          '';
+        };
```
Comment (sumnerevans) [inline modules/programs/comodoro.nix:63]: Are there any good defaults for this or will users have to specify both of these values to run the server?

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/programs/comodoro.nix (hunk 3 above)
Comment (soywod) [reply modules/programs/comodoro.nix:63]: These values are mandatory ones, hence the `nonEmpty*`.

File: modules/services/comodoro.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          "PASSWORD_STORE_DIR" = "~/.password-store";
+        }
+      '';
+      description = ''
+        Extra environment variables to be exported in the service.
+      '';
+    };
+
+    settings = {
+      preset = lib.mkOption {
```
Comment (rycee) [inline modules/services/comodoro.nix:36]: I don't think  `settings` is necessary here. It is not describing an RFC-42 style configuration like in the programs module.

```suggestion
    preset = lib.mkOption {
```

File: tests/modules/programs/comodoro/default.nix [hunk 5]
```diff
@@ -0,0 +1 @@
+{ comodoro = ./comodoro.nix; }
```
Comment (rycee) [inline tests/modules/programs/comodoro/default.nix:1]:
```suggestion
{ comodoro-program = ./comodoro.nix; }
```

**Comment by rycee** [comment]
Thanks! Added some additional comments. Also please fix the commit  message. Should be

> comodoro: add module

**Comment by rycee** [comment]
Thanks! Merged to master now.

**Pull request merged**